gstreamer/src/pad.rs

```diff
@@ -1113,15 +1113,15 @@ unsafe extern "C" fn trampoline_query_function(
 ) -> glib_ffi::gboolean {
     let _guard = CallbackGuard::new();
     #[cfg_attr(feature = "cargo-clippy", allow(transmute_ptr_to_ref))]
-    let func: &&(Fn(&Pad, &Option<::Object>, &mut ::QueryRef) -> bool
+    let func: &&(Fn(&Pad, &Option<::Object>, ::Query) -> bool
                          + Send
                          + Sync
                          + 'static) = transmute((*pad).querydata);
 
     func(
         &from_glib_borrow(pad),
         &from_glib_borrow(parent),
-        ::QueryRef::from_mut_ptr(query),
+        from_glib_full(query),
```

Comment:
This is wrong, same problem as with the pad probe. We don't own the query, we only borrow it mutably.